examples/message.js

```diff
@@ -0,0 +1,22 @@
+var phantom = require("../lib");
+var _ph, _page, _outObj;
+
+phantom.create().then(ph => {
+    _ph = ph;
+    return _ph.createPage();
+}).then(page => {
+    _page = page;
+    _ph.onMessage(function(resourceUrl) {
+    	console.log('Node context received resourceUrl: ', resourceUrl);
+    });
+    _page.property('onResourceRequested', function(resource, networkRequest, callbackMessagePrefix) {
+    	console.log(callbackMessagePrefix + resource.url);
```

Comment:
I have a feeling you would have to put some it in `shim.js`. Phantomjs runs that file so it should be possible to do it that way. But when I think about the end user writing code, it just sounds confusing. :/